src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -46,25 +46,25 @@ class TaskStatusUpdateProcessorImpl @Inject() (
     taskTracker.task(taskId).flatMap {
       case Some(task) if task.launched.isDefined =>
         val taskStateOp = TaskStateOp.MesosUpdate(task, MarathonTaskStatus(status), now)
-        stateOpProcessor.process(taskStateOp).flatMap(_ => acknowledge(status))
+        stateOpProcessor.process(taskStateOp).flatMap(_ => acknowledge(status, ack))
 
       case maybeTask: Option[Task] if killWhenUnknownOrNotLaunched(status) =>
         killUnknownTaskTimer {
           val taskStr = taskKnownOrNotStr(maybeTask)
           log.warn(s"Kill $taskStr $taskId")
           killTask(taskId.mesosTaskId)
-          acknowledge(status)
+          acknowledge(status, ack)
         }
 
       case maybeTask: Option[Task] =>
         val taskStr = taskKnownOrNotStr(maybeTask)
         log.info(s"Ignoring ${status.getState} update for $taskStr $taskId")
-        acknowledge(status)
+        acknowledge(status, ack)
     }
   }
 
-  private[this] def acknowledge(taskStatus: MesosProtos.TaskStatus): Future[Unit] = {
-    driverHolder.driver.foreach(_.acknowledgeStatusUpdate(taskStatus))
+  private[this] def acknowledge(taskStatus: MesosProtos.TaskStatus, ack: Boolean = true): Future[Unit] = {
```

Comment:
I think we introduced the ack only for the "invented" TaskStatusUpdate. Do we still need this?